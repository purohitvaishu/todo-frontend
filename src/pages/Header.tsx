/* eslint-disable no-unused-vars */
/* eslint-disable @typescript-eslint/no-unused-vars */
import React from "react";
import { makeStyles } from "@material-ui/core/styles";
import { connect } from "react-redux";
import Typography from "@material-ui/core/Typography";
import { Divider, Grid, Button } from "@material-ui/core";
import AddIcon from "@material-ui/icons/Add";
import Modal from "../components/Modal";
import { callPopup } from "../actions/popup.actions";
import { State } from "../interfaces/state.interface";
import { ActionCall } from "../interfaces/actions.interface";

const useStyles = makeStyles(theme => {
  return {
    root: {
      padding: theme.spacing(1)
    },
    divider: {
      marginTop: 24
    },
    heading: {
      display: "flex",
      justifyContent: "center"
    },
    topHead: {
      fontWeight: 600,
      color: "darkcyan"
    },
    button: {
      textAlign: "center",
      marginTop: "8%"
    }
  };
});

type Props = State & ActionCall;

const Header: React.FC<Props> = (props: Props) => {
  const classes = useStyles();
  const { open, callPopup } = props;

  const handleClickOpen = () => {
    if (callPopup) callPopup({}, !open, false);
  };

  return (
    <>
      <Grid style={{ display: "flex" }}>
        <Grid item md={5} sm={12} className={classes.divider}>
          <Divider />
        </Grid>
        <Grid item md={2} sm={12} className={classes.heading}>
          <Typography variant="h4" component="h2" style={{ fontWeight: 600 }}>
            Tasks&nbsp;
          </Typography>
          <Typography variant="h4" component="h2" className={classes.topHead}>
            Lists
          </Typography>
        </Grid>
        <Grid item md={5} sm={12} className={classes.divider}>
          <Divider />
        </Grid>
      </Grid>
      <Grid className={classes.button}>
        <Button variant="outlined" onClick={handleClickOpen}>
          <AddIcon />
        </Button>
        <Typography variant="h5" component="h2">
          Add List
        </Typography>
      </Grid>
      {open && <Modal />}
    </>
  );
};

const mapStateToProps = state => {
  return {
    open: state.popup.open
  };
};

const mapDispatchToProps = {
  callPopup
};

export default connect(mapStateToProps, mapDispatchToProps)(Header);
