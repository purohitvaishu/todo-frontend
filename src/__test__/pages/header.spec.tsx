/* eslint-disable react/jsx-props-no-spreading */
import React from "react";
import { shallow } from "enzyme";
import Header from "../../pages/Header";

describe("Test: Header", () => {
  test("container", () => {
    const props = {
      callPopup: jest.fn(),
      open: false
    };

    const result = shallow(<Header.WrappedComponent {...props} />);
    expect(result.debug()).toMatchSnapshot();
  });
});
