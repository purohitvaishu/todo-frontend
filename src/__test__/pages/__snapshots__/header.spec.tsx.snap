// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test: Header container 1`] = `
"<Fragment>
  <WithStyles(ForwardRef(Grid)) style={{...}}>
    <WithStyles(ForwardRef(Grid)) item={true} md={5} sm={12} className=\\"makeStyles-divider-2\\">
      <WithStyles(ForwardRef(Divider)) />
    </WithStyles(ForwardRef(Grid))>
    <WithStyles(ForwardRef(Grid)) item={true} md={2} sm={12} className=\\"makeStyles-heading-3\\">
      <WithStyles(ForwardRef(Typography)) variant=\\"h4\\" component=\\"h2\\" style={{...}}>
        TasksÂ 
      </WithStyles(ForwardRef(Typography))>
      <WithStyles(ForwardRef(Typography)) variant=\\"h4\\" component=\\"h2\\" className=\\"makeStyles-topHead-4\\">
        Lists
      </WithStyles(ForwardRef(Typography))>
    </WithStyles(ForwardRef(Grid))>
    <WithStyles(ForwardRef(Grid)) item={true} md={5} sm={12} className=\\"makeStyles-divider-2\\">
      <WithStyles(ForwardRef(Divider)) />
    </WithStyles(ForwardRef(Grid))>
  </WithStyles(ForwardRef(Grid))>
  <WithStyles(ForwardRef(Grid)) className=\\"makeStyles-button-5\\">
    <WithStyles(ForwardRef(Button)) variant=\\"outlined\\" onClick={[Function: handleClickOpen]}>
      <Memo([object Object]) />
    </WithStyles(ForwardRef(Button))>
    <WithStyles(ForwardRef(Typography)) variant=\\"h5\\" component=\\"h2\\">
      Add List
    </WithStyles(ForwardRef(Typography))>
  </WithStyles(ForwardRef(Grid))>
</Fragment>"
`;
